src/devicemanager/devicemanager.cpp

```diff
@@ -151,19 +152,57 @@ void DeviceManager::updateAvailableConnections(
     const QVector<LinkConfiguration>& availableLinkConfigurations, const QString& detector)
 {
     qCDebug(DEVICEMANAGER) << "Available devices:" << availableLinkConfigurations;
-    // Make all connections unavailable by default
+
     for (int i {0}; i < _sensors[Available].size(); i++) {
-        auto linkConf = _sensors[Connection][i].value<QSharedPointer<LinkConfiguration>>();
+        const auto linkConf = _sensors[Connection][i].value<QSharedPointer<LinkConfiguration>>();
         if (linkConf->isSimulation() || _sensors[DetectorName][i] != detector) {
             continue;
         }
+
+        // Make all connections unavailable by default
         _sensors[Available][i] = false;
         const auto indexRow = index(i);
         emit dataChanged(indexRow, indexRow, _roles);
     }
 
+    // Check if the configuration already exists for a sensor
+    // Serial ports does not support multiple devices connected
+    // Some sensors, like Ping1D, can fail to answer a ABR procedure
     for (const auto& link : availableLinkConfigurations) {
-        append(link, PingHelper::nameFromDeviceType(link.deviceType()), detector);
+        const bool sameSerialDevice = std::any_of(
+            _sensors[Connection].cbegin(), _sensors[Connection].cend(), [&link](const QVariant& variantLink) {
+                const auto sensorLink = variantLink.value<QSharedPointer<LinkConfiguration>>().get();
+                qCDebug(DEVICEMANAGER) << "Device" << sensorLink
+                                       << "already already provided by a different connection:" << link;
+                return link.serialPort() == sensorLink->serialPort() && link != *sensorLink;
+            });
+
+        if (!sameSerialDevice) {
+            append(link, PingHelper::nameFromDeviceType(link.deviceType()), detector);
+        }
+    }
+
+    // We'll let the link to fail the communication attempt "a max number of fails" before making it unavailable
+    // This is necessary to avoid any problem related to automatic baud rates problem from the sensor side.
+    static const int maxNumberOfFails = 3;
+    for (int i {0}; i < _sensors[Available].size(); i++) {
+        const auto linkConf = _sensors[Connection][i].value<QSharedPointer<LinkConfiguration>>();
+        if (linkConf->isSimulation()) {
+            continue;
+        }
+
+        const auto indexRow = index(i);
```

Comment:
explicit types when it's ipossible to guess the type by reading the code?

Reply:
`indexRow = index(i)` appears to be pretty clear that is an index value. We should also not care what type it's, since it's an argument for `dataChanged`.